### PR TITLE
Feat: redux flow integration

### PR DESCRIPTION
**Pull request by BastinJafari**
Setup a redux store and connect the Flowview to it

### CODE REVIEW

**Comment by deepankarm** [comment]
Should we merge `package-lock.json` file? It is >23K lines long.

**Review by imsergiy — CHANGES_REQUESTED** [review]

**Comment by BastinJafari** [comment]
> Should we merge `package-lock.json` file? It is >23K lines long.

I had this conversation with Roshan in Slack:

Bastin Jafari  12:53 PM
What do you think about adding the package-lock.jon to gitignore? As far as I know, this get's calculated anyway from package.json

Roshan Jossy  12:58 PM
I'm against it.
When someone runs npm install, npm will look at versions in package-lock.json if it's available and not package.json.
I think yarn introduced this first for javascript.
This makes sure that they also get the same versions we have and that gives a higher probability that it'll work the same way.
Versions in package.json are flexible. It could be higher than this version or variable patch versions.

12:58
Also, to make sure that it works the same way in CI as in our local, lock file is neccessary.

**Comment by deepankarm** [comment]
> > Should we merge `package-lock.json` file? It is >23K lines long.
> 
> I had this conversation with Roshan in Slack:
> 
> Bastin Jafari 12:53 PM
> What do you think about adding the package-lock.jon to gitignore? As far as I know, this get's calculated anyway from package.json
> 
> Roshan Jossy 12:58 PM
> I'm against it.
> When someone runs npm install, npm will look at versions in package-lock.json if it's available and not package.json.
> I think yarn introduced this first for javascript.
> This makes sure that they also get the same versions we have and that gives a higher probability that it'll work the same way.
> Versions in package.json are flexible. It could be higher than this version or variable patch versions.
> 
> 12:58
> Also, to make sure that it works the same way in CI as in our local, lock file is neccessary.

Cool. Here's a [thread](https://stackoverflow.com/questions/44206782/do-i-commit-the-package-lock-json-file-created-by-npm-5) that mentions the same.

File: src/redux/flows/flows.types.ts [hunk 1]
```diff
@@ -14,54 +19,44 @@ export type LoadFlowAction = {
...
-      needs?: Object;
-      send_to?: Object;
-      depth?: number;
-    };
+
+export interface Node extends Omit<INode, "type"> {
+  label?: string;
+  needs?:
+    | {
+        encode: boolean;
```
Comment (imsergiy) [inline src/redux/flows/flows.types.ts:27]: why does this interface specifically define `encode`?

Comment (BastinJafari) [reply src/redux/flows/flows.types.ts:27]: It shouldn't :)
Weird that it works.

I would change it to [pod: string]: boolean

Also now that I see it in 51 from pea to pod, which would be more accurate

Comment (BastinJafari) [reply src/redux/flows/flows.types.ts]: Changed it

**Review by imsergiy — APPROVED** [review]

**Pull request merged**